mava/wrappers/jaxmarl.py

```diff
@@ -403,3 +410,42 @@ def get_global_state(self, wrapped_env_state: BraxState, obs: Dict[str, Array])
         """Get global state from observation and copy it for each agent."""
         # Use the global state of brax.
         return jnp.tile(wrapped_env_state.obs, (self.num_agents, 1))
+
+
+class MPEWrapper(JaxMarlWrapper):
+    """Wrapper for the MPE environment."""
+
+    def __init__(
+        self,
+        env: SimpleSpreadMPE,
+        has_global_state: bool = False,
+    ):
+        super().__init__(env, has_global_state, env.max_steps)
+        self._env: SimpleSpreadMPE
+
+    @cached_property
+    def action_dim(self) -> chex.Array:
+        "Get the actions dim for each agent."
+        # Adjusted automatically based on the action_type specified in the kwargs.
+        if _is_discrete(self._env.action_space(self.agents[0])):
+            return self._env.action_space(self.agents[0]).n
+        return self._env.action_space(self.agents[0]).shape[0]
+
+    @cached_property
+    def state_size(self) -> chex.Array:
+        "Get the state size of the global observation"
+        # The shape provided in the `observation_spaces` isn't always the same as
+        # that given after constructing the full agent's obs
+        obs, _ = self._env.reset(jax.random.PRNGKey(0))
+        return obs["agent_0"].shape[0] * self.num_agents
+
+    def action_mask(self, wrapped_env_state: BraxState) -> Array:
+        """Get action mask for each agent."""
+        return jnp.ones((self.num_agents, self.action_dim), dtype=bool)
+
+    def get_global_state(self, wrapped_env_state: BraxState, obs: Dict[str, Array]) -> Array:
+        """Get global state from observation and copy it for each agent."""
+
+        all_obs = jnp.array([obs[agent] for agent in self._env.agents]).flatten()
+        all_obs = jnp.expand_dims(all_obs, axis=0).repeat(self._env.num_agents, axis=0)
+        return all_obs
```

Comment:
Just double checking do they provide a global state?

Reply:
Nope and they used the classic method in this wrapper ([world_state](https://github.com/FLAIROx/JaxMARL/blob/42e7d638aa5dab88757c95c22f0486af8ec85694/baselines/MAPPO/mappo_rnn_mpe.py#L61)) for their MAPPO baseline.